src/not_haskell.erl

```diff
@@ -0,0 +1,30 @@
+-module(not_haskell).
+
+-export([mapAccumL/3, for/2]).
+
+
+-spec mapAccumL(Fun, Acc, List) -> {Acc, List} when
+  %% boo args are backwards... why...
+  Fun :: fun((Elem :: T, Acc) -> {Acc, Out}),
+  Acc :: term(),
+  List :: [T],
+  Out :: term(),
+  T :: term().
+mapAccumL(Fun, Acc, List) ->
+  lists:foldl(fun(_x, {_acc, _outlist}) ->
+    {NAcc, NOut} = Fun(_x, _acc),
+    %% inefficient snocing, w/e
+    %% TODO implement using foldr instead
```

Comment:
you can also reverse the list first

Reply:
:o

Reply:
in haskell foldl' is implemented using foldr by folding a function and evaluating it. foldr' is implemented in the same way using foldl

    foldl' :: (b -> a -> b) -> b -> t a -> b
    foldl' f z0 xs = foldr f' id xs z0
      where f' x k z = k $! f z x

Reply:
and obv. this is how foldl is implemented :scream: 

 foldl :: (b -> a -> b) -> b -> t a -> b
    foldl f z t = appEndo (getDual (foldMap (Dual . Endo . flip f) t)) z

Reply:
whats the difference between foldl and foldl'? 😱

Reply:
foldl' is strict version of foldl.
The ! after the $ makes the function application strict.

Reply:
what's a strict function application? 😱 😱

Reply:
not lazy :sleepy:

Reply:
neat, didn't realize that was a thing